### ISSUE TITLE
Sdl2 OpenGL flag always being used

### ISSUE THREAD

**Issue by bootzin** [mellinoe/veldrid#222]
Hey Mellino,

I recently started fiddling around your code because I'm trying to make my own game engine on .NET Core (mostly for learning purposes) and decided to use Veldrid as my graphics library.

As I was making my research on Window creation, I was looking at your code on the VeldridStartup class and I've realized that you always pass the SDL_WindowFlags.OpenGL as a flag to the window creation, even when not using the OpenGL backend. Is there a reason behind this?

I've read an unanswered and somewhat old post on stack overflow saying that this flag made SDL_CreateWindow() take about 10 times more time to finish ([source](https://stackoverflow.com/questions/42776455/slow-window-creation-with-sdl-window-opengl-flag)), so maybe it could also be related to #216?

**Comment by mellinoe** [comment]
I doubt that it is related to #216, but this is an interesting point overall. The reason the flag is currently unconditionally passed is to allow an OpenGL GraphicsDevice to be created targeting it later on. I'd like to see some stats about whether this flag does actually slow things down (a simple benchmark would be fine), but it seems like an additional overload somewhere could prevent this flag from being passed if it does represent a performance hit.

**Comment by bootzin** [comment]
Right, I'll see if I can get to do some testing this weekend then, and I'll share the results.

On a side note, the Resize event  is being fired multiple times for the Sdl2Window. After investigating, I saw that you're sending Resize events on the HadleWindowEvent on multiple different 'cases'. So when we resize the window, we actually get a call for Resize because of the resized event, but we also get a call for Resize because of the SizeChanged event. If we maximize the window we actually get three calls instead.. Wouldn't it be better if differente events were called for each of these? Or maybe abstract it so that the resize event is only called once... Because having to treat this in the applicaiton itself feels a little shady

Anyway, not criticizing, I just really liked Veldrid and want it to be as reliable as possible, because I really think it is a nice project

**Comment by mellinoe** [comment]
@bootzin Regarding the side note: I have a fix for that on a local branch. I agree that it's an issue and that there should not be duplicate callbacks for a single logical resize event. NeoDemo and other similar programs sidestep this issue a little bit by delaying the Swapchain resize until the beginning of a frame (thus de-duping the callbacks in a sense), but it shouldn't behave that way in the first place. I'll work on getting that branch into master sometime soon.
